test/metabase/api/action_test.clj

```diff
@@ -113,54 +113,62 @@
+                                     (-> (assoc :response_handle ".body.result"  :description nil))))
+                  expected-fn    (fn [m]
+                                   (cond-> m
+                                     (= (:type initial-action) "implicit")
+                                     (assoc :parameters [{:id "id" :type "type/BigInteger" :special "hello"}])))
+                  updated-action (update-fn initial-action)]
+              (testing "Create fails with"
+                (testing "no permission"
                   (is (= "You don't have permissions to do that."
-                         (mt/user-http-request :rasta :post 403 "action" initial-action)))))
-              (testing "Update"
-                (is (partial= (expected-fn updated-action)
-                              (mt/user-http-request :crowberto :put 200 action-path (update-fn {}))))
-                (testing "Should not be possible without permission"
-                  (is (= "You don't have permissions to do that."
-                         (mt/user-http-request :rasta :put 403 action-path (update-fn {}))))))
-              (testing "Get"
-                (is (partial= (expected-fn updated-action)
-                              (mt/user-http-request :crowberto :get 200 action-path)))
-                (testing "Should not be possible without permission"
-                  (is (= "You don't have permissions to do that."
-                         (mt/user-http-request :rasta :get 403 action-path)))))
-              (testing "Get All"
-                (is (partial= [{:id exiting-implicit-action-id, :type "implicit", :kind "row/update"}
-                               (expected-fn updated-action)]
-                              (mt/user-http-request :crowberto :get 200 (str "action?model-id=" card-id))))
-                (testing "Should not be possible without permission"
-                  (is (= "You don't have permissions to do that."
-                         (mt/user-http-request :rasta :get 403 (str "action?model-id=" card-id))))))
-              (testing "Delete"
-                (testing "Should not be possible without permission"
-                  (is (= "You don't have permissions to do that."
-                         (mt/user-http-request :rasta :delete 403 action-path))))
-                (is (nil? (mt/user-http-request :crowberto :delete 204 action-path)))
-                (is (= "Not found." (mt/user-http-request :crowberto :get 404 action-path)))))))))))
+                         (mt/user-http-request :rasta :post 403 "action" initial-action))))
+                (when (= "query" (:type initial-action))
+                  (testing "actions disabled"
+                    (mt/with-actions-disabled
+                      (is (= "Actions are not enabled."
+                             (:cause
+                              (mt/user-http-request :crowberto :post 400 "action" initial-action))))))))
```

Comment:
This is the only new test. Apologies for the very noisy diff; the tests were very fragile about when they expected `created-action` to exist. I'm not thrilled about it, but I suppose it's more efficient than creating a new action for each assertion.